### DISCUSSION TITLE
Feature Request: Prioritization of Music Providers

### DISCUSSION THREAD

**Discussion by kamilkosek** [music-assistant/support#2994] [Feature requests and ideas]
**Description**: I would like to request a feature that allows users to prioritize music providers when selecting tracks. For example, if a user selects a playlist from Spotify, the system should first check if each track is available from higher-priority providers (as defined by the user) before falling back to Spotify. This would ensure that users can listen to tracks from their preferred providers even if the playlist source is from a lower-priority provider.

**Use Case:**

User has a playlist on Spotify but prefers to stream music from a higher-priority provider, such as Tidal or Apple Music.
The system should automatically play the track from the highest-priority provider available in the user's list, regardless of the playlist's source.

**Benefits:**

Allows users to leverage multiple music services more effectively.
Enhances user control over music streaming preferences.
Minimizes disruptions from service limitations (e.g., quality differences or availability across providers).
Thank you for considering this feature request

**Comment by OzGav** [comment]
MA always streams the highest quality available. https://music-assistant.io/installation/?h=quality#usage-and-notes

**Reply by kamilkosek** [reply]
Ah thanks, smh. I´ve overseen this. 
And are there any options to prefer local files or Jellyfin  over remote providers?

**Reply by OzGav** [reply]
Have a look again at the docs 

> local is preferred over cloud
